virttest/utils_libvirt/libvirt_ceph_utils.py

```diff
@@ -0,0 +1,154 @@
+    device_bus = params.get("virt_disk_device_bus", "virtio")
+    device = params.get("virt_disk_device", "disk")
+    device_target = params.get("virt_disk_device_target", "vdb")
+    hotplug = "yes" == params.get("virt_disk_device_hotplug", "no")
+    keep_raw_image_as = "yes" == params.get("keep_raw_image_as", "no")
+
+    # Ceph related configurations
+    ceph_mon_ip = params.get("ceph_mon_ip", "EXAMPLE_MON_HOST")
+    ceph_host_port = params.get("ceph_host_port", "EXAMPLE_PORTS")
+    ceph_disk_name = params.get("ceph_disk_name", "EXAMPLE_SOURCE_NAME")
+    ceph_client_name = params.get("ceph_client_name")
+    ceph_client_key = params.get("ceph_client_key")
+    ceph_auth_user = params.get("ceph_auth_user")
+    ceph_auth_key = params.get("ceph_auth_key")
+    auth_sec_usage_type = params.get("ceph_auth_sec_usage_type", "ceph")
+    storage_size = params.get("storage_size", "1G")
+    img_file = params.get("ceph_image_file")
+    key_file = os.path.join(data_dir.get_tmp_dir(), "ceph.key")
+    key_opt = ""
+    is_local_img_file = True if img_file is None else False
+
+    # Prepare a blank params to confirm if delete the configure at the end of the test
+    ceph_cfg = ""
+    disk_auth_dict = None
+    if not utils_package.package_install(["ceph-common"]):
+        raise exceptions.TestError("Failed to install ceph-common")
+
+    # Create config file if it doesn't exist
+    ceph_cfg = ceph.create_config_file(ceph_mon_ip)
+    # If enable auth, prepare a local file to save key
+    if ceph_client_name and ceph_client_key:
+        with open(key_file, 'w') as f:
+            f.write("[%s]\n\tkey = %s\n" %
+                    (ceph_client_name, ceph_client_key))
+        key_opt = "--keyring %s" % key_file
+    if is_setup:
+        # If enable auth, prepare device source
+        if ceph_client_name and ceph_client_key:
+            auth_sec_dict = {"sec_usage": auth_sec_usage_type,
+                             "sec_name": "ceph_auth_secret"}
+            auth_sec_uuid = libvirt.create_secret(auth_sec_dict)
+            virsh.secret_set_value(auth_sec_uuid, ceph_auth_key,
+                                   debug=True)
+            disk_auth_dict = {"auth_user": ceph_auth_user,
+                              "secret_type": auth_sec_usage_type,
+                              "secret_uuid": auth_sec_uuid}
+            device_source = "rbd:%s:mon_host=%s:keyring=%s" % (ceph_disk_name,
+                                                               ceph_mon_ip,
+                                                               key_file)
+        else:
+            device_source = "rbd:%s:mon_host=%s" % (ceph_disk_name, ceph_mon_ip)
+        logging.debug("device source is: %s", device_source)
+        # clean up image file if exists
+        cmd = ("rbd -m {0} {1} info {2} && rbd -m {0} {1} rm "
+               "{2}".format(ceph_mon_ip, key_opt, ceph_disk_name))
+        process.run(cmd, ignore_status=True, shell=True, verbose=True)
```

Comment:
Same as above

Reply:
need add verbose=True, do you mean this ?

Reply:
No, I'd like to propose to call ceph.rbd_image_rm.

Reply:
You might be familiar with rbd command lines, but it might be not easy for others to understand these shell command lines(eg, L92~94 & L142~144). Since the function already exists and can be used, I think the proper way is to try to use the existing way instead of repeating the operations by executing the shell command lines.

BTW, can you please reply the comment instead of edit mine? So I'll get a notification...

Reply:
1)That is quote reply, not edit your comment.
2)process.run(verbose=True) implemented above will output both command line and result.  Don't forget process.run itself is one utility.
3)Better not assume that others should  use existed utility if there is one existed. It depends on some factors, such as robustness or flexability, complexisity the utility have.  in many cases, one utility can be called 'nice utility' if 80% scenarios can be covered.
4)Specifically to this utility, it separate rbd_pool_name, rbd_image_name as two parameters .Practically   in our cfg ,we only provides "pool_name/img_name" as one whole parameter. Do we need painfully split the parameters and rewrap them? Moreover , namespace parameter which rbd_image has , I don't think in rbd command area ,we have an mapping concept for this.

Reply:
@dzhengfy @kylazhang @chloerh @smitterl @kamilvarga Would you mind sharing your ideas?

Reply:
I agree with using the existing function instead of running cmd. If we do not use it, then why we make it as common function for ceph?

Reply:
> I agree with using the existing function instead of running cmd. If we do not use it, then why we make it as common function for ceph?

@kylazhang 
see previous explanation

Reply:
I also agree with using existing function instead of hardcoding the commands in every ceph related tp-libvirt cases. I remember I did same suggestion before in some ceph related cases before.

Reply:
> I also agree with using existing function instead of hardcoding the commands in every ceph related tp-libvirt cases. I remember I did same suggestion before in some ceph related cases before.

@dzhengfy 
it is ceph utiliy to be exposed ,not specific tp-libvirt cases
rbd command is changed to ceph.rbd_image_rm ,except that convert command,which is not supported in current ceph.py